### ISSUE TITLE
GAT Link prediction for inductive learning

### ISSUE THREAD

**Issue by Yaxin-Cui** [stellargraph/stellargraph#1939]
### GAT link prediction on unseen nodes

*Hi! I am trying to implement link prediction based on GAT and predict unseen nodes (Inductive learning). While I can successfully run a link prediction using GAT on the network with the same number of nodes, when new nodes come, it can not work properly. I am wondering whether this is caused by the GAT link generator (FullBatchLinkGenerator)*

###  The code: (notice train and test networks have different number of nodes)
`                        train_gen = FullBatchLinkGenerator(G_train,method="gat")
                        test_gen = FullBatchLinkGenerator(G_test,method="gat")
                        
                        gat = GAT(layer_sizes=[8, 4],
                        activations=["elu", "softmax"],
                        attn_heads=8,
                        bias = True,
                        generator=train_gen,
                        in_dropout=0.5,
                        attn_dropout=0.5,
                        normalize=None)

                        x_inp, x_out = gat.in_out_tensors()

                        prediction = LinkEmbedding(activation="relu",method ="ip")(x_out)
                        prediction = keras.layers.Reshape((-1,))(prediction)

                        #Train tf model for classification
                        model = keras.Model(inputs=x_inp, outputs=prediction)
                        
                        model.compile(optimizer=keras.optimizers.Adam(lr=1e-3), loss=keras.losses.binary_crossentropy, metrics=[keras.metrics.AUC()])
                        
                        
                        
                        init_train_metrics = model.evaluate(train_gen.flow(edge_ids_train,edge_labels_train))
                        init_test_metrics = model.evaluate(test_gen.flow(edge_ids_test, edge_labels_test))`

The error occurs when evaluating the model: 

> Cannot multiply A and B because inner dimension does not match: 387 vs. 403.  Did you forget a transpose?  Dimensions of A: [387, 387).  Dimensions of B: [403,8]

Screenshots: 
![image](https://user-images.githubusercontent.com/64431841/122833083-f8ca8b80-d2b1-11eb-8978-13a14f0a9826.png)

**Comment by YINTONG0090** [comment]
This problem also occurred when I used the GCN and SAGE models, and it seemed that the internal structure of the function was set up incorrectly...
